crypto3/libs/zk/include/nil/crypto3/zk/snark/systems/plonk/placeholder/lookup_argument.hpp

```diff
@@ -173,268 +119,138 @@ namespace nil {
                             prepare_lookup_input(mask_assignment, lagrange0);
                         auto& lookup_input = *lookup_input_ptr;
 
-
-                        // 3. Lookup_input and lookup_value are ready
-                        //    Now sort them!
-                        //    Reduce value and input:
+                        // Lookup_input and lookup_value are ready
+                        // Reduce value and input, count how many times lookup inputs appear in lookup values.
                         auto reduced_value_ptr = std::make_unique<std::vector<polynomial_dfs_type>>();
                         auto& reduced_value = *reduced_value_ptr;
 
-                        for( std::size_t i = 0; i < lookup_value.size(); i++ ){
+                        for (std::size_t i = 0; i < lookup_value.size(); i++) {
                             reduced_value.push_back(reduce_dfs_polynomial_domain(lookup_value[i], basic_domain->m));
                         }
+
                         auto reduced_input_ptr = std::make_unique<std::vector<polynomial_dfs_type>>();
                         auto& reduced_input = *reduced_input_ptr;
 
-                        for( std::size_t i = 0; i < lookup_input.size(); i++ ){
+                        for (std::size_t i = 0; i < lookup_input.size(); i++) {
                             reduced_input.push_back(reduce_dfs_polynomial_domain(lookup_input[i], basic_domain->m));
                         }
 
-                        //    Sort
-                        auto sorted = sort_polynomials(reduced_input, reduced_value, basic_domain->m,
-                            preprocessed_data.common_data->desc.usable_rows_amount);
+                        // Compute the counts of how many times a lookup input appears in the lookup values.
+                        std::vector<polynomial_dfs_type> counts = count_lookup_input_appearances(
+                            reduced_input, reduced_value, basic_domain->m, usable_rows_amount);
+
+                        // Commit to the counts.
+                        commitment_scheme.append_to_batch(LOOKUP_BATCH, counts);
```

Comment:
May be added to VARIABLE_VALUES_BATCH to decrease number of batches.

Reply:
Created tasks for all proposals:

https://www.notion.so/nilfoundation/LogUp-Commit-counts-polynomial-into-the-VARIABLE_VALUES_BATCH-batch-removing-the-LOOKUP_BATCH-1a8c614852608071b892dd73bc97c795

https://www.notion.so/nilfoundation/LogUp-Group-lookup-inputs-before-inversion-based-on-some-threshold-of-maximal-degree-This-will-red-1a8c61485260806aaf73f0ca3c95d1ca

https://www.notion.so/nilfoundation/LogUp-Implement-Logarithmic-Derivatives-in-the-permutation-argument-and-remove-unnecessary-values--19ec61485260801bb249d2ea9fda05f4